src/openms_gui/source/VISUAL/SpectraViewWidget.cpp

```diff
@@ -199,115 +206,80 @@ namespace OpenMS
       emit spectrumDoubleClicked(spectrum_index);
     }
     else
-    {
-      // open several chromatograms at once
-      std::vector<int> chrom_indices;
-      for (Int i = 0; i != res.size(); ++i)
-      {
-        chrom_indices.push_back(res[i].toInt());
-      }
-      emit spectrumDoubleClicked(chrom_indices);
+    { // open several chromatograms at once
+      emit spectrumDoubleClicked(listToVec(res));
     }
 
   }
 
-  void SpectraViewWidget::spectrumContextMenu_(const QPoint & pos)
+  void SpectraViewWidget::spectrumContextMenu_(const QPoint& pos)
   {
     QTreeWidgetItem * item = spectra_treewidget_->itemAt(pos);
     if (item)
     {
       //create menu
       int spectrum_index = item->text(1).toInt();
-      QMenu * context_menu = new QMenu(spectra_treewidget_);
-      context_menu->addAction("Show in 1D view");
-      context_menu->addAction("Meta data");
-      context_menu->addAction("Center here");
-
-      QAction * selected = context_menu->exec(spectra_treewidget_->mapToGlobal(pos));
-      if (selected != nullptr && selected->text() == "Show in 1D view")
+      QMenu context_menu(spectra_treewidget_);
+      context_menu.addAction("Show in 1D view", [&]()
       {
         std::vector<int> chrom_indices;
-        const QList<QVariant> & res = item->data(0, 0).toList();
+        const QList<QVariant>& res = item->data(0, 0).toList();
         if (res.size() == 0)
         {
           emit showSpectrumAs1D(spectrum_index);
         }
         else
-        {
-          // open several chromatograms at once
-          for (Int i = 0; i != res.size(); ++i)
-          {
-            chrom_indices.push_back(res[i].toInt());
-          }
-          emit showSpectrumAs1D(chrom_indices);
+        { // open several chromatograms at once
+          emit showSpectrumAs1D(listToVec(res));
```

Comment:
chromatogram handling is still suboptimal and something we might need to look at in the future

Reply:
that would be awesome!